ql/instruments/bonds/amortizingfixedratebond.hpp

```diff
@@ -44,7 +44,10 @@ namespace QuantLib {
                                 const Period& exCouponPeriod = Period(),
                                 const Calendar& exCouponCalendar = Calendar(),
                                 BusinessDayConvention exCouponConvention = Unadjusted,
-                                bool exCouponEndOfMonth = false);
+                                bool exCouponEndOfMonth = false,
+                                const std::vector<Real>& redemptions = { 100.0 },
+                                Natural PaymentLag = 0);
```

Comment:
There are already a bunch of default parameters before these ones, so it doesn't matter much, but I think `paymentLag` is more likely to be passed than `redemptions`, isn't it?  If so, I'd change the order.

Reply:
Also, in what case is the payment lag needed?  I've seen it mentioned in swaps but not in bonds so far.

Reply:
The paymentLag in this situation is probably more appropriately referred to as Delay Days - in US MBS for example, borrowers payments for the accrual period in the prior calendar month are passed through to investors on the 25th of the payment. It is this delay that paymentLag is supposed to capture. TBH I don't quite understand the role of the already existing argument in the amortizingFloatingBond constructor?

Reply:
Fixing days?  Those would affect the index fixing; for instance, a coupon starting on Wednesday 20 would use as rate the Euribor fixing of Monday 18, 2 days before.

Reply:
I think there is a paymentLag arg in the amortizingFloatingRateBond constructor - that is what I was referring to.

Reply:
It was added by this pull request.

Reply:
Ah, OK - I got my signals crossed with the aborted attempts. So in both cases they act as delay days between the end of the accrual period and when the cashflow is actually received.